### DISCUSSION TITLE
Schema import in taxonomy

### DISCUSSION THREAD

**Discussion by cjundt** [JeffFerguson/gepsio#49] [Q&A]
Hi,

I'm working on new ESRS taxonomy (https://xbrl.efrag.org/taxonomy/draft-esrs/2023-07-31/esrs_all.xsd), which is using xsd:import feature. This is not implemented in Gepsio and thus most of the label linkbases are missing.
I'm not shure how to implement this because XmlSchema is already partly handling it, have you any idea ?

**Comment by JeffFerguson** [comment]
Good question! I am not sure. Let me take a look at the schema you've provided and start looking into this. Thanks for the tip!

**Reply by cjundt** [reply]
Hello,

I've been investigating a bit on the subject, I think the best would be to use the compiled XmlSchemaSet features to read annotations et parse linkbases : 
```C#
	foreach( XmlSchema schema in thisSchemaSet.Schemas( ) ) {
		foreach( var annotation in schema.Items.OfType<XmlSchemaAnnotation>(  ) ) {
			foreach( var item in annotation.Items ) {
				...
			}
		}
	}
```

Because, compiled XmlSchemaSet seems to handle properly the imports.

**Comment by JeffFerguson** [comment]
I can't think of a reason to say "no" ... what are the implications for any issues around breaking changes? I suppose that you could create a branch off of `develop` to play around with the idea ...

**Comment by JeffFerguson** [comment]
I will turn this question into an issue so that it can be tracked as such.

**Comment by JeffFerguson** [comment]
Please continue the discussion on the issue at https://github.com/JeffFerguson/gepsio/issues/50.
